src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -590,16 +590,18 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                         diagnostics);
 
                 case BoundKind.IndexOrRangePatternIndexerAccess:
-                    var patternIndexer = (BoundIndexOrRangePatternIndexerAccess)expr;
-                    // If we got here this should be a pattern indexer taking a Range,
+                    var implicitIndexerAccess = (BoundIndexOrRangePatternIndexerAccess)expr;
+                    Debug.Assert(implicitIndexerAccess.IndexerAccess is BoundCall);
+                    // If we got here this should be an implicit indexer taking a Range,
                     // meaning that the pattern symbol must be a method (either Slice or Substring)
-                    return CheckMethodReturnValueKind(
-                        (MethodSymbol)patternIndexer.PatternSymbol,
-                        patternIndexer.Syntax,
-                        node,
-                        valueKind,
-                        checkingReceiver,
-                        diagnostics);
+                    return CheckValueKind(node, implicitIndexerAccess.IndexerAccess, valueKind, checkingReceiver, diagnostics);
+
+                case BoundKind.IndexOrRangeIndexerPatternReceiverPlaceholder:
+                    var receiverPlaceholder = (BoundIndexOrRangeIndexerPatternReceiverPlaceholder)expr;
+                    return CheckValueKind(node, receiverPlaceholder.Expression, valueKind, checkingReceiver, diagnostics);
+
+                case BoundKind.DeconstructValuePlaceholder:
+                    break;
```

Comment:
>break; [](http://example.com/codeflow?start=20&length=6)

Is this code path covered by tests?

Reply:
It's covered by `DeconstructRefExtensionMethod`. The cases the only need to use the expression as an r-value are handled above (line 418, `RequiresRValueOnly` case).